### PR TITLE
Feature: Filter games by category

### PR DESCRIPTION
**Pull request by orende**
## Description

This PR addresses issue #543.

The PR adds a dialog that can be used to filter the games library based on game categories. Enabling one category filters shows only games matching the selected category, while enabling multiple category filters shows games matching any of the selected categories.
The games and associated categories are stored on disk in the user's cache directory to allow for filtering in offline mode. Once the client goes online again, the file will be updated with any new games and categories found.

Note that this solution uses the `category` field as returned by the call to `embed.gog.com/account/getFilteredProducts` and not the `genres` field as returned by the call to `gamesdb.gog.com/platforms/gog/external_releases`. This is because the former requires only a single API call, while the latter initially requires one API call per game in the library. 

Searching for categories, as proposed in the issue, was left out as the total number of categories was low (8 in total).

The PR also replaces all print statements with the logger library for ease of debugging.

## Checklist
 
 - [x] _CHANGELOG.md_ was updated (**format**: - Change made (thanks to github_username))

### CODE REVIEW

File: data/ui/application.ui [hunk 1]
```diff
@@ -141,6 +141,27 @@
...
+            <property name="can_focus">True</property>
+            <property name="receives_default">False</property>
+            <property name="tooltip_text" translatable="yes" context="category_filter" comments="Tooltip for category filter button">Category filters</property>
+            <property name="valign">center</property>
+            <signal name="clicked" handler="on_menu_category_filter_clicked" swapped="no"/>
+            <child>
+              <object class="GtkImage">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="icon_name">filter-symbolic</property>
```
Comment (sharkwouter) [inline data/ui/application.ui:156]: This icon is not included into Ubuntu 22.04 and Debian 12 by default, so we'll need to use another icon.

File: minigalaxy/ui/categoryfilters.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from minigalaxy.ui.gtk import Gtk
+
+
+@Gtk.Template.from_file(os.path.join(UI_DIR, "categoryfilters.ui"))
+class CategoryFilters(Gtk.Dialog):
+    __gtype_name__ = "CategoryFilters"
+
+    genre_filtering_grid = Gtk.Template.Child()
+    filter_dict = {}
+    categories = [
```
Comment (sharkwouter) [inline minigalaxy/ui/categoryfilters.py:16]: Are these the standard categories for GOG? Do you have a source for these?

**Review by sharkwouter — CHANGES_REQUESTED** [review]

**Comment by sharkwouter** [comment]
Looks good, thanks for this! I've found the icon does not load, otherwise I'd merge it.

File: data/ui/application.ui (hunk 1 above)
Comment (orende) [reply data/ui/application.ui:156]: Okay, I'll change it. Is there a list somewhere of available icons?

File: minigalaxy/ui/categoryfilters.py (hunk 2 above)
Comment (orende) [reply minigalaxy/ui/categoryfilters.py:16]: The source for these categories is the list in the lower left of this GOG store page: https://www.gog.com/en/games

image

File: data/ui/application.ui (hunk 1 above)
Comment (orende) [reply data/ui/application.ui]: I've replaced the icon with `view-conceal-symbolic` which I found in the `/usr/share/icons/Adwaita` folder (Adwaita being a Gnome theme, so hopefully that icon should be included in stock Ubuntu and Debian). Let me know if that one works.

**Comment by sharkwouter** [comment]
Thanks! That works!

**Pull request merged**